subsys/net/l2/ethernet/ethernet.c

```diff
@@ -986,12 +986,8 @@ struct device *net_eth_get_ptp_clock(struct net_if *iface)
 	}
 
 	return api->get_ptp_clock(net_if_get_device(iface));
-#else
-	return NULL;
-#endif
 }
 
-#if defined(CONFIG_PTP_CLOCK)
```

Comment:
Looks like "net: ptp: clock: Create a dummy inline func for net_eth_get_ptp_clock()" should have gone first, then you wouldn't need to add flip-flop add/remove code to the codebase.